tailnet/conn.go

```diff
@@ -387,16 +387,17 @@ func (c *Conn) Closed() <-chan struct{} {
 // Close shuts down the Wireguard connection.
 func (c *Conn) Close() error {
 	c.mutex.Lock()
-	defer c.mutex.Unlock()
 	select {
 	case <-c.closed:
+		c.mutex.Unlock()
 		return nil
 	default:
 	}
+	close(c.closed)
+	c.mutex.Unlock()
 	for _, l := range c.listeners {
```

Comment:
Do we need to do some guarding wrt `(*Conn).Listen` as well? Wondering if there could be a race modifying this map.

Reply:
@mafredri good find. I amended the commit to add a check there. Let me know if it's what you were thinking!

Reply:
Yup, looks good! 👌🏻